server/sonar-server/src/main/java/org/sonar/server/computation/scm/Changeset.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.scm;
+
+import java.util.Objects;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public final class Changeset {
+
+  private final String revision;
+  private final long date;
+  @CheckForNull
+  private final String author;
+
+  private Changeset(Builder builder) {
+    this.revision = builder.revision;
+    this.author = builder.author;
+    this.date = builder.date;
+  }
+
+  public static Builder newChangesetBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private String revision;
+    private long date;
+    @CheckForNull
+    private String author;
+
+    private Builder() {
+      // prevents direct instantiation
+    }
+
+    public Builder setRevision(String revision) {
+      this.revision = checkRevision(revision);
+      return this;
+    }
+
+    public Builder setDate(long date) {
+      this.date = date;
+      return this;
+    }
+
+    public Builder setAuthor(@Nullable String author) {
+      this.author = author;
+      return this;
+    }
+
+    public Changeset build() {
+      checkRevision(revision);
+      return new Changeset(this);
+    }
+
+    private static String checkRevision(String revision){
+      return requireNonNull(revision, "Revision cannot be null");
+    }
+
+  }
+
+  public String getRevision() {
+    return revision;
+  }
+
+  public long getDate() {
+    return date;
+  }
+
+  @CheckForNull
+  public String getAuthor() {
+    return author;
+  }
+
+  @Override
+  public boolean equals(@Nullable Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Changeset changeset = (Changeset) o;
+    return Objects.equals(revision, changeset.revision) &&
```

Comment:
since revision can not be `null` and date is primitive, you could save two calls to `Objects.equals`